src/io/file_descriptor.windows.cr

```diff
@@ -194,90 +144,44 @@ class IO::FileDescriptor
   end
 
   private def unbuffered_read(slice : Bytes)
-    count = slice.size
-    loop do
-      bytes_read = LibC.read(@fd, slice.pointer(count).as(Void*), count)
-      if bytes_read != -1
-        return bytes_read
-      end
-
-      if Errno.value == Errno::EAGAIN
-        wait_readable
-      else
-        raise Errno.new "Error reading file"
-      end
-    end
-  ensure
-    if (readers = @readers) && !readers.empty?
-      add_read_event
+    overlapped = GC.malloc(sizeof(LibWindows::Overlapped)).as(LibWindows::Overlapped*)
+    overlapped.value.status = 0
+    overlapped.value.bytes_transfered = 0
+    overlapped.value.offset = @pos
+    overlapped.value.event = nil
+
+    if LibWindows.read_file(@handle, slice.pointer(slice.size), slice.size, out bytes_read, overlapped)
+      @pos += bytes_read
+      return bytes_read
+    elsif LibWindows.get_last_error == WinError::ERROR_HANDLE_EOF
+      return 0
+    elsif LibWindows.get_last_error == WinError::ERROR_IO_PENDING
+      raise WinError.new "ReadFile: TODO, Implement Overlapped write"
+    else
+      raise WinError.new "ReadFile"
     end
   end
 
   private def unbuffered_write(slice : Bytes)
     count = slice.size
     total = count
     loop do
-      if LibWindows.write_file(@fd, slice.pointer(count), count, out bytes_written, nil)
+      overlapped = GC.malloc(sizeof(LibWindows::Overlapped)).as(LibWindows::Overlapped*)
```

Comment:
Same here. It could also be created outside the loop for reuse:
```crystal
overlapped = LibWindows::Overlapped.new

loop do
  # reset members
  overlapped.status = 0
  overlapped.bytes_transferred = 0
  overlapped.offset = @pos

  if LibWindows.write_file(..., out bytes_written, pointerof(overlapped))
    @pos += bytes_written
  end
end
```